libraries/chain/dbs_budget.cpp

```diff
@@ -204,13 +260,62 @@ bool dbs_budget::_check_autoclose(const budget_object& budget)
 {
     if (budget.balance.amount <= 0)
     {
-        close_budget(budget);
+        _close_budget(budget);
         return true;
     }
     else
     {
         return false;
     }
 }
+
+bool dbs_budget::_is_fund_budget(const budget_object& budget) const
+{
+    return budget.owner == SCORUM_ROOT_POST_PARENT;
+}
+
+void dbs_budget::_close_budget(const budget_object& budget)
+{
+    if (_is_fund_budget(budget))
+    {
+        _close_fund_budget(budget);
+    }
+    else
+    {
+        _close_owned_budget(budget);
+    }
+}
+
+void dbs_budget::_close_owned_budget(const budget_object& budget)
+{
+    FC_ASSERT(!_is_fund_budget(budget), "not allowed for fund budget");
+
+    dbs_account& account_service = db().obtain_service<dbs_account>();
+
+    const auto& owner = account_service.get_account(budget.owner);
+
+    // withdraw all balance rest asset back to owner
+    //
+    asset repayable = budget.balance;
+    if (repayable.amount > 0)
+    {
+        // check if input budget state != budget in DB
```

Comment:
Does this comment is valid?

Reply:
Consequence of constant rework of logic. I will fix it